run_kraken.py

```diff
@@ -152,6 +152,60 @@ def post_actions(kubeconfig_path, scenario, failed_post_scenarios, pre_action_ou
     return failed_post_scenarios
 
 
+def pod_scenarios(scenarios_list, config):
+    try:
+        # Loop to run the scenarios starts here
+        for pod_scenario in scenarios_list:
+            if len(pod_scenario) > 1:
+                pre_action_output = run_post_action(kubeconfig_path, pod_scenario[1])
+            else:
+                pre_action_output = ''
+            runcommand.invoke("powerfulseal autonomous --use-pod-delete-instead-of-ssh-kill"
+                              " --policy-file %s --kubeconfig %s --no-cloud"
+                              " --inventory-kubernetes --headless"
+                              % (pod_scenario[0], kubeconfig_path))
+
+            logging.info("Scenario: %s has been successfully injected!" % (pod_scenario[0]))
+            logging.info("Waiting for the specified duration: %s" % (wait_duration))
+            time.sleep(wait_duration)
+
+            failed_post_scenarios = post_actions(kubeconfig_path, pod_scenario,
+                                                 failed_post_scenarios, pre_action_output)
+            publish_kraken_status(config, failed_post_scenarios)
```

Comment:
Currently you an error when running the first: `scenarios/etcd.yml. Encountered the following exception: local variable 'failed_post_scenarios' referenced before assignment`
I think here we need to return the failed_post_scenarios so that they can be rerun later if they do fail. This would only have the last failure I believe. 

failed_post_scenarios also needs to be added as an argument and appended to instead of resetting.

Reply:
Ah, sorry. I have updated the code, PTAL. Thank you!

Reply:
Looks awesome thanks so much!!